drivers/accel/adxl367/adxl367.c

```diff
@@ -0,0 +1,1310 @@
+{
+	int32_t ret;
+	uint8_t temp[2] = {0, 0};
+	uint8_t reg_data, nready = 1U;
+
+	while (nready) {
+		ret = adxl367_get_register_value(dev, &reg_data, ADXL367_REG_STATUS, 1);
+		if (ret)
+			return -1;
+		if (reg_data & ADXL367_STATUS_DATA_RDY)
+			nready = 0U;
+	}
+
+	ret = adxl367_get_register_value(dev,
+					 temp,
+					 ADXL367_REG_TEMP_H,
+					 2);
+	if (ret)
+		return ret;
+
+	*raw_temp = ((int16_t)temp[0] << 6) + (temp[1] >> 2);
+	// extend sign to 16 bits
+	if (*raw_temp & NO_OS_BIT(13))
+		*raw_temp |= NO_OS_GENMASK(15, 14);
+
+	return 0;
+}
+
+/***************************************************************************//**
+ * @brief Converts raw temperature value to Celsius degree value.
+ *
+ * @param dev       - The device structure.
+ * @param raw_temp  - Raw temperature value.
+ * @param temp   	- Struct to store converted data.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+*******************************************************************************/
+static int32_t adxl367_temp_conv(struct adxl367_dev *dev,
+				 uint16_t raw_temp, struct adxl367_fractional_val* temp)
+{
+	int64_t temp_data;
+
+	temp_data = (int16_t)raw_temp;
+
+	temp_data += ADXL367_TEMP_OFFSET;
+	temp_data *= ADXL367_TEMP_SCALE;
+
+	temp->integer = no_os_div_s64_rem(temp_data,
+					  ADXL367_TEMP_SCALE_DIV,
+					  &(temp->fractional));
+
+	return 0;
+}
+
+/***************************************************************************//**
+ * @brief Reads the temperature of the device.
+ *
+ * @param dev  - The device structure.
+ * @param temp - Temperature output data.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t adxl367_read_temperature(struct adxl367_dev *dev,
+				 struct adxl367_fractional_val *temp)
+{
+	int16_t raw_temp;
+	int32_t ret;
+	int64_t dividend;
+
+	ret = adxl367_read_raw_temp(dev, &raw_temp);
+	if (ret)
+		return ret;
+
+	return adxl367_temp_conv(dev, raw_temp, temp);
+}
+
+/***************************************************************************//**
+ * @brief Reads ADC data. If ADXL367_ADC_EN is not set, use
+ * 	adxl367_adc_read_en() first to enable ADC reading.
+ *
+ * @param dev      - The device structure.
+ * @param data 	   - ADC value.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t adxl367_read_adc(struct adxl367_dev *dev, int16_t *data)
+{
+	int32_t ret;
+	uint8_t raw[2] = {0, 0};
+	uint8_t reg_data, nready = 1U;
+
+	while (nready) {
+		ret = adxl367_get_register_value(dev, &reg_data, ADXL367_REG_STATUS, 1);
+		if (ret)
+			return -1;
+		if (reg_data & ADXL367_STATUS_DATA_RDY)
+			nready = 0U;
+	}
+
+	ret = adxl367_get_register_value(dev,
+					 raw,
+					 ADXL367_REG_EX_ADC_H,
+					 2);
+	if (ret)
+		return ret;
+
+	*data = ((int16_t)raw[0] << 6) + (raw[1] >> 2);
```

Comment:
Would it be an idea to consider  using a function for this operation, since it is used in several places?